### PR TITLE
updated react and react-dom

### PR DESCRIPTION
**Pull request by 0x00b1**
Hi, @JackUrb! I needed to update react and react-dom for the compression [task](https://github.com/fairinternal/mephisto-internal/pull/21). Let me know if this works. 

cc: @mmuckley

### CODE REVIEW

**Comment by JackUrb** [comment]
@pringshia any thoughts on how we deal with react versions in Mephisto? Has been an issue for other projects as well

**Comment by pringshia** [comment]
This PR includes quite a bit of package version updates which I'm hesitant on merging right away, as there could be unintended behavior. Tests are breaking as well. To back up a bit, what part of the prior version of `react` / `react-dom` was causing issues? Also I would advise to break down the PR to introduce the updates across the various packages incrementally instead of all-at-once.

**Comment by 0x00b1** [comment]
> To back up a bit, what part of the prior version of react / react-dom was causing issues? Also I would advise to break down the PR to introduce the updates across the various packages incrementally instead of all-at-once.

Lack of `useDeferredValue` and `useTransition`. Also the ability to use @mui/material 5.10.4.

File: .github/workflows/annotated-pkgs.yml [hunk 1]
```diff
@@ -24,9 +24,9 @@ jobs:
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/setup-node@v3
```
Comment (pringshia) [inline .github/workflows/annotated-pkgs.yml:27]: This is causing tests to fail due to the following issue: https://github.com/actions/setup-node/issues/317

**Comment by pringshia** [comment]
Given the node version bump on the runners, seems like we'll also need to [update yarn to version 3.2.2+](https://github.com/yarnpkg/berry/issues/4778) to fix the issue seen [here](https://github.com/facebookresearch/Mephisto/runs/8275418328?check_suite_focus=true):

```
➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR node:internal/errors:477
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     ErrorCaptureStackTrace(err);
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     ^
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR 
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR Error [ERR_LOADER_CHAIN_INCOMPLETE]: "file:///home/runner/work/Mephisto/Mephisto/.pnp.loader.mjs 'resolve'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     at new NodeError (node:internal/errors:387:5)
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     at ESMLoader.resolve (node:internal/modules/esm/loader:[85](https://github.com/facebookresearch/Mephisto/runs/8275528832?check_suite_focus=true#step:9:89)2:13)
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     at async ESMLoader.getModuleJob (node:internal/modules/esm/loader:431:7)
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     at async Promise.all (index 0)
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     at async ESMLoader.import (node:internal/modules/esm/loader:533:24)
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     at async loadESM (node:internal/process/esm_loader:91:5)
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR     at async handleMainPromise (node:internal/modules/run_main:65:12) {
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR   code: 'ERR_LOADER_CHAIN_INCOMPLETE'
  ➤ YN0000: │ @parcel/watcher@npm:2.0.4 STDERR }
  ➤ YN0009: │ @parcel/watcher@npm:2.0.4 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-d4a4833f/build.log)
```

**Comment by pringshia** [comment]
Following up on this, I think you should just be able to relax the peerDependency requirement for `mephisto-task` instead of doing such a broad version bump across the codebase. For example: https://github.com/palantir/blueprint/blob/develop/packages/core/package.json#L63-L64

**Comment by pringshia** [comment]
Resolved with #905

**Pull request closed without merging**